components/remote_settings/src/client.rs

```diff
@@ -450,6 +436,83 @@ impl Client {
     }
 }
 
+/// Stores all the endpoints for a Remote Settings server
+///
+/// There's actually not to many of these, so we can just pack them all into a struct
+struct RemoteSettingsEndpoints {
+    /// Top-level URL for Remote Settings server
+    ///
+    /// This has the form `https://[domain]/v1/`. It's where we get the attachment base url from.
+    server_url: Url,
```

Comment:
In other implementations, we call this the root URL and it has the form of `{server_url}/`